Allow for use of podman for running containers in development.

We migrated the generation of the python sdk code to use docker.  Currently, we require that docker must be enabled on host to build.  

Kubernetes/Kind actually allow the use of podman rather than docker.  

It would be ideal to allow for either Podman or Docker for running the test code.

```
kehannon@kehannon-thinkpadp1gen4i:~/Work/jobset$ make test
test -s /home/kehannon/Work/jobset/bin/controller-gen && /home/kehannon/Work/jobset/bin/controller-gen --version | grep -q v0.11.4 || \
GOBIN=/home/kehannon/Work/jobset/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.11.4
/home/kehannon/Work/jobset/bin/controller-gen \
        rbac:roleName=manager-role output:rbac:artifacts:config=config/components/rbac\
        crd:generateEmbeddedObjectMeta=true output:crd:artifacts:config=config/components/crd/bases\
        webhook output:webhook:artifacts:config=config/components/webhook\
        paths="./..."
go fmt ./...
go vet ./...
test -s /home/kehannon/Work/jobset/bin/setup-envtest || GOBIN=/home/kehannon/Work/jobset/bin go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
echo "testing Python SDK..."
testing Python SDK...
./hack/python-sdk/test-sdk.sh
ERROR: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
make: *** [Makefile:130: test-python-sdk] Error 1
```

--- a/hack/python-sdk/test-sdk.sh
+++ b/hack/python-sdk/test-sdk.sh
@@ -19,9 +19,11 @@ repo_root="$(dirname "${BASH_SOURCE}")/../.."
 
 cd "${repo_root}/sdk/python"
 
+# Allow one to use podman or docker for local testing.
+CONTAINER_ENGINE=${CONTAINER_ENGINE:-docker}
 ## For CI we found that docker wasn't started.
 ## Should be a no-op if docker is up
 
 ## If non ubuntu machine, install docker in your path
-docker buildx build -f Dockerfile -t python-unit .
-docker run python-unit pytest test
+${CONTAINER_ENGINE} buildx build -f Dockerfile -t python-unit .
+${CONTAINER_ENGINE} run python-unit pytest test

